### DISCUSSION TITLE
corfu + dabbrev autocompletion auto popup

### DISCUSSION THREAD

**Discussion by wjj1928** [minad/corfu#519] [Q&A] [answered]
I try below config, when I write below text:
```
xyzabc
xyzbcd
xyz
```
When my cursor at the third line after `z`, then I need press `M-/` to popup the completion list.

What I expected is after I type the `z`, the completion list popup automatically.
Which config should be added for such purpose?

```
use-package corfu
  :ensure t
  :custom
  (corfu-auto t)
  (corfu-cycle t)
  (corfu-quit-at-boundary 'separator)
  (corfu-quit-no-match 'separator)
  (corfu-scroll-margin 0)
  (corfu-separator ?\s)
  (eldoc-add-command #'corfu-insert)
  (corfu-echo-documentation 0)
  (corfu-popupinfo-delay '(0.5 . 0.2))
  (corfu-auto-prefix 1)
   (corfu-on-exact-match nil)
  :config
  (global-corfu-mode)
  (corfu-history-mode)
  (corfu-popupinfo-mode +1)
  (use-package dabbrev
    :bind (("M-/" . dabbrev-completion)
           ("C-M-/" . dabbrev-expand)))
    (use-package kind-icon
		 :ensure t
    :custom
    (kind-icon-default-face 'corfu-default) ;; Compute blended background correctly
    :config
    (add-to-list 'corfu-margin-formatters #'kind-icon-margin-formatter))
  :hook (eglot-managed-mode . corfu-mode))

```

**Comment by jdtsmith** [comment, marked as answer]
Note that `corfu` does not control which completions are provided; your setup does.  Dabbrev is not a CAPF, but it does call `completion-at-point`, which means `corfu` is utilized for its interface.  To achieve what you want, you could use `cape` — specifically `cape-capf-super` — to composite your mode's normal CAPF with `cape-dabbrev`.  

The other thing worth noting is that at any point you can _switch_ to dabbrev completions just by hitting `M-/`.  I have recently found that more useful.
